src/ussd_clinic_rapidpro.js

```diff
@@ -140,16 +140,62 @@ go.app = function() {
                         );
                     }
                 },
-                next: "state_get_contact"
+                next: "state_clinic_code"
             });
         });
 
+        self.add("state_clinic_code", function(name, opts) {
+            var text;
+            // Run a no-wait contact check in the background to populate the cache
+            var msisdn = utils.normalize_msisdn(
+                _.get(self.im.user.answers, "state_enter_msisdn", self.im.user.addr), "ZA");
+            self.whatsapp.contact_check(msisdn, false).then(_.noop, _.noop);
+            if(!self.im.user.answers.on_whatsapp) {
+                return self.states.create("state_not_on_whatsapp");
+            }
```

Comment:
I don't see anywhere that would set this `on_whatsapp` answer at this point, it only gets set later in the flow, so this condition will always trigger, and the use will always be sent to `state_not_on_whatsapp`